### PR TITLE
Policy engine feed mappers for Microsoft feeds

### PR DESCRIPTION
**Pull request by nightfurys**
Signed-off-by: Swathi Gangisetty <swathi@anchore.com>

**What this PR does / why we need it**:
Policy engine logic for syncing Microsoft vulnerability and product data

**Which issue this PR fixes** *(optional, in `fixes #<issue number>)(, fixes #<issue_number, ...)` format, will close the issue when PR is merged*: fixes #:

**Special notes**:

### CODE REVIEW

File: anchore_engine/services/policy_engine/engine/feeds/feeds.py [hunk 1]
```diff
@@ -733,6 +734,53 @@ class GithubFeed(VulnerabilityFeed):
...
+
+        if group_obj.name == MicrosoftFeed._products_group:
+            count = db.query(GenericFeedDataRecord).filter(GenericFeedDataRecord.group == group_obj.name).delete()
+            log.info(log_msg_ctx(operation_id, group_obj.name, group_obj.feed_name, 'Flushed {} group records'.format(count)))
+            group_obj.last_sync = None  # Null the update timestamp to reflect the flush
+            group_obj.count = 0
+        else:
+            # VulnerabilityFeed.__flush_helper_fn__(db=db, feed_name=group_obj.feed_name, group_name=group_obj.name)
+
+            count = db.query(FixedArtifact).filter(FixedArtifact.namespace_name == group_obj.name).delete()
```
Comment (zhill) [inline anchore_engine/services/policy_engine/engine/feeds/feeds.py:762]: These shouldn't be necessary, the relationship cascades will handle it. Only the Vulnerability object delete is needed to handle both artifact flushes.

**Comment by zhill** [comment]
I'm going to close and re-open this to kick it in CI since it wasn't building forked PRs

**Comment by zhill** [comment]
Looks like it needs to be rebased on master to pick up the fixes for the idna package version conflict. That should resolve the build failure.

**Comment by nightfurys** [comment]
Will rebase and resubmit the PR. Would it be easier if the PR is not created from my fork?

**Comment by zhill** [comment]
> Will rebase and resubmit the PR. Would it be easier if the PR is not created from my fork?

No, the circle config is correct now to build from forks so it shouldn't matter

File: anchore_engine/services/policy_engine/engine/feeds/feeds.py (hunk 1 above)
Comment (nightfurys) [reply anchore_engine/services/policy_engine/engine/feeds/feeds.py]: Fixed

**Comment by nightfurys** [comment]
Closing this PR out, to be merged in later

**Pull request closed without merging**